lab.md

```diff
@@ -0,0 +1,498 @@
+
+In our `index controller`, let's add the following code:
+
+```js
+angular
+.module("grumblr")
+.controller("GrumbleIndexController", [
+  GrumbleIndexControllerFunction
+]);
+
+function GrumbleIndexControllerFunction(){
+  console.log("I'm in the controller!");
+}
+```
+
+Link to this file in `index.html`
+
+```html
+<!-- after app.js -->
+<script src='js/grumbles/index.controller.js'></script>
+```
+
+</section>
+
+
+## Use Index controller when url is `/grumbles`
+
+<section markdown="1">
+
+```js
+// app.js
+
+function RouterFunction($stateProvider){
+  $stateProvider
+  .state("grumbleIndex", {
+    url: "/grumbles",
+    controller: "GrumbleIndexController",
+    controllerAs: "GrumbleIndexViewModel"
+  });
+}
+```
+
+
+</section>
+
+## Create an element with `ui-view` in index.html
+
+<section markdown="1">
+
+```html
+<!-- ./index.html -->
+<div ui-view></div>
+```
+Visit: http://localhost:9000/#/grumbles
+
+You should see "I'm in the controller!" in the console.
+
+</section>
+
+## Load Template when url is `/grumbles`
+
+<section markdown="1">
+
+We can have Angular load and insert whole HTML files for us -- just like with *partials* in Rails.
+
+Let's create a folder in which we can put some partials:
+
+```
+$ touch js/grumbles/index.html
+```
+
+Let's put a piece of HTML into that `index.html`, just so we know it's working:
+
+```html
+<h2>I'm the Grumbles index!</h2>
+```
+
+```js
+function RouterFunction($stateProvider){
+  $stateProvider
+  .state("grumbleIndex", {
+    url: "/grumbles",
+    controller: "GrumbleIndexController",
+    controllerAs: "GrumbleIndexViewModel"
+    templateUrl: "js/grumbles/index.html"
+  });
+}
+```
+
+</section>
+
+## When visiting `/grumbles` in url:
+
+<section markdown="1">
+
+- You should see "I'm in the controller" logged to the console.
+- You should see "I'm the grumbles index" in the browser
+
+Things to check:
+
+- Did you link to all your js files in index.html
+- Are there any errors in the console?
+
+```html
+<script src="js/app.js"></script>
+<script src="js/grumbles/index.controller.js"></script>
+```
+</section>
+
+## Load fake data into index controller
+
+<section markdown="1">
+
+Before `<script src="js/app.js">`, let's add this:
+
+```html
+    <script>
+      var grumbles = [
+        {
+          title: "I am Grumble One"
+        },
+        {
+          title: "I'm another Grumble"
+        }
+      ]
+    </script>
+    <script src="js/app.js"></script>
+```
+
+We can access this global variable in all the other files. Set `this.grumbles` equal to that variable in our controller:
+
+```js
+// js/grumbles/index.controller.js
+
+function GrumbleIndexControllerFunction(){
+  this.grumbles = grumbles;
+}
+```
+
+</section>
+
+## Loop through data and display info about each grumble
+
+<section markdown="1">
+
+```html
+<!-- js/grumbles/index.html -->
+<div data-ng-repeat="grumble in GrumbleIndexViewModel.grumbles">
+  <p>{{grumble.title}}</p>
```

Comment:
Weird but I think these braces were not escaped...the entire expression doesn't show up on the rendered html